packages/webapp/src/components/ActionMenu/index.tsx

```diff
@@ -0,0 +1,75 @@
+/*
+ *  Copyright 2024 LiteFarm.org
+ *  This file is part of LiteFarm.
+ *
+ *  LiteFarm is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU General Public License as published by
+ *  the Free Software Foundation, either version 3 of the License, or
+ *  (at your option) any later version.
+ *
+ *  LiteFarm is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ *  GNU General Public License for more details, see <https://www.gnu.org/licenses/>.
+ */
+
+import { ReactNode } from 'react';
+import clsx from 'clsx';
+import TextButton from '../Form/Button/TextButton';
+import styles from './styles.module.scss';
+
+interface action {
+  label: string;
+  onClick: () => void;
+}
+
+interface iconAction extends action {
+  icon: ReactNode;
+}
+
+export interface ActionMenuProps {
+  headerLeftText?: string;
+  textActions?: action[];
+  iconActions: iconAction[];
+  classes?: { root?: string };
+}
+
+const ActionMenu = ({
+  headerLeftText,
+  textActions = [],
+  iconActions,
+  classes = {},
+}: ActionMenuProps) => {
+  // If the className is not yet supported, ensure to add corresponding styles for the widths
+  // of the component and each iconGroup to the SCSS file.
+  const iconCountClassName = styles[`iconCount_${iconActions.length}`];
```

Comment:
I think we can make this work more generically without having to add a specific class and do the math for each number of icons, but it was kinda hard to explain in words so I branched off and tried it out in a branch of my own. I pushed it, the branch name is`generic-styling-for-action-menu` -- take a look and let me know what you think!

Reply:
Thank you for reviewing! I don't see any commits in your branch, have you pushed new commits??
The width of the action menu needs to be `408px` for 4 icon buttons, so I added calculation but look forward to seeing your solution!

Reply:
I didn't push 😅 lol! Just pushed them

Reply:
Thank you @antsgar!
I took a look at your code. I think the problems that we will have when not specifying the widths are:
1. Without the component width, the component will get longer when label texts are long. (Also I followed [Loïc's design](https://www.figma.com/file/eMffC0h3JPlWtzNttxJpil?node-id=3093:35169&mode=design#708694966) that the component width should be `408px` for four buttons.)
2. Without the width for each button, the space between buttons differ depending on the label texts.
![Screenshot 2024-02-26 at 9 49 27 AM](https://github.com/LiteFarmOrg/LiteFarm/assets/33141219/876f3a03-a1a0-4d88-b2dd-5779ea4edbea)

What do you think?

Reply:
Ah, I see! In that case your approach seems good since it's truer to the design spec, let's keep that 🙂